### DISCUSSION TITLE
Include an Invite Link generator in the People > Invitations area of Admin Portal

### DISCUSSION THREAD

**Discussion by ellativity** [forem/forem#16327] [Feature Requests]
Building on from https://github.com/forem/forem/issues/16226 to suggest that this might also allow us to have an Invite Link Generator tool in the invitations section.

Rather than entering email addresses and sending invites, Admins could create a time-limited link (maybe with the option to select the validity time like Discord?), and share the invite with people they don't have email addresses for. 

Use cases include:

- inviting people from DEV, Discord, Facebook, or other existing community spaces where emails are not shared; 
- allowing people to choose their own email address to sign up from rather than being bound to the email address they were invited with;
- allow for data analysis: Invite Links could have a listed history to allow Admins to see when a specific link was generated, how many people joined with it, etc.

**Comment by ildi-x** [comment]
I really like this suggestion! It would also be cool if an admin can include a customizable "invite caption" with each invite link. I can see myself creating personalized captions for different users. This might be a nice surprise and fun experience when joining a Forem community. 

I also think that when creating multiple unique invite links, there needs to be a method of being able to know which link a new users registered with in order to deal with abuse in the event that a link falls in the wrong hands. If admins notice that a specific invite link is resulting in fake accounts being created, then a decision can be made to terminate/revoke that particular invite link.

**Reply by ellativity** [reply]
I love both of these Admin insights, @ildi-x (and the new username 😉). There's definitely a line to draw between data collection for security management purposes (necessary) and data collection for targeted advertising or identitification.

Especially when you think that the invite code is only necessary in instances where a Forem is invite-only, it makes sense for the Admin to know if a code is being abused.

**Reply by ildi-x** [reply]
The more I think about it, the more excited I get about the invite link generator idea. I know its prob not on the prority list of new features, but it would be nice to have. 

Im so glad my username change didn't break anything, im still new to github 😇
